core/shared/src/main/scala/zio/scheduler/Scheduler.scala

```diff
@@ -16,9 +16,51 @@
 
 package zio.scheduler
 
-import zio.internal.{ Scheduler => IScheduler }
-import zio.{ Has, ZLayer }
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.TimeUnit
+
+import zio.duration.Duration
+import zio.{ UIO, ZLayer }
 
 object Scheduler extends PlatformSpecific {
-  val live: ZLayer.NoDeps[Nothing, Has[IScheduler]] = ZLayer.succeed(globalScheduler)
+  private[zio] type CancelToken = () => Boolean
+
+  trait Service extends Serializable {
+    def schedule[A](task: => A, duration: Duration): UIO[A] =
```

Comment:
Maybe we should even change this to be `task: ZIO[R, E, A]` in order to be "totally pure"? If we're intending effects to be placed here, we should think about calling it `effectSchedule` to be consistent.

Do we need the effectful version or is the pure version enough?

Reply:
Yes! It is very easy to put an effect in there accidentally and then your effect won't do anything because it isn't being run. It looks like `schedule` is really only being used to power `sleep` so the "totally pure" version should be fine.